devlib/utils/ssh.py

```diff
@@ -205,9 +205,12 @@ def execute(self, command, timeout=None, check_exit_code=True,
         try:
             with self.lock:
                 _command = '({}); __devlib_ec=$?; echo; echo $__devlib_ec'.format(command)
-                raw_output = self._execute_and_wait_for_prompt(
-                    _command, timeout, as_root, strip_colors)
-                output, exit_code_text, _ = raw_output.rsplit('\r\n', 2)
+                split_output = self._execute_and_wait_for_prompt(
+                    _command, timeout, as_root, strip_colors).rsplit('\r\n', 2)
+                try:
+                    output, exit_code_text, _ = split_output
+                except ValueError as e:
+                    raise TargetStableError("cannot parse ssh command reply - target misconfiguration?")
```

Comment:
Maybe save unsplit output and report it here, as part of the error?

Reply:
yes, you're right that could be useful.  i considered that but decided not to, because of the effort involved in sanitising it.

it might contain unprintable characters and it might be very long, so, i think the correct approach would be to

`raise TargetStableError("cannot parse ssh reply (target misconfiguration: {}".format(limit_length(80, hexdump(split_output)))`

i shouldn't be logged separately, because it is used to determine certain features (`needs_root`, IIRC)

if you think it's worthwhile, i'll implement it that way - ?

Reply:
I would just filter out or encode unprintable characters, rather than produce hexdump, as the common case is that the output will contain text.

Do not limit the length at this stage -- report the whole thing; it's up to the code handling and/or reporting the error to perform formatting such as length limiting.

Reply:
in that case the code handling it can also do the conversion, i'd say...

Reply:
Yeah, I'm cool wit that.